utils/posts.ts

```diff
@@ -0,0 +1,37 @@
+import { extract } from "std/encoding/front_matter/yaml.ts";
+import { join } from "std/path/mod.ts";
+
+export interface Post {
+  slug: string;
+  title: string;
+  publishedAt: Date;
+  content: string;
+  abstract: string;
+}
+
+export async function getPosts(): Promise<Post[]> {
+  const files = Deno.readDir("./routes/blog");
+  const promises = [];
+  for await (const file of files) {
+    if (file.name.endsWith(".tsx")) continue;
+    const slug = file.name.replace(".md", "");
+    promises.push(getPost(slug));
+  }
+  const posts = await Promise.all(promises) as Post[];
+  posts.sort((a, b) => b.publishedAt.getTime() - a.publishedAt.getTime());
+  return posts;
+}
+
+export async function getPost(slug: string): Promise<Post | null> {
+  const text = await Deno.readTextFile(
+    join("./routes/blog", `${slug}`, `index.md`),
```

Comment:
Can we have blog post files named `[slug].md` instead of `[slug]/index.md`? Is there a particular reason for having the post files as `index.md` within a `[slug]` parent folder?

Reply:
Of course we can do that. I was thinking about was the possibility to use pictures in the articles. But when I'm thinking more about this case I would say that those pictures and assets should land in the `static` folder anyway. I will change it to `[slug].md`